pygmt/clib/conversion.py

```diff
@@ -95,9 +97,12 @@ def dataarray_to_matrix(grid):
         coord_incs = coord[1:] - coord[0:-1]
         coord_inc = coord_incs[0]
```

Comment:
If the coordinate increments can be slightly different, is it safe to just take the first one, or should we take the mean/average value (inside the `if not np.allclose(coord_incs, coord_inc):` block)?

Reply:
Yes, using the mean spacing sounds better.

Reply:
Or we can calculate the increment from the minimum, maximum and length of the coordinates? Take the coordinates in https://github.com/GenericMappingTools/pygmt/pull/1530#discussion_r713752741 as an example, the array `grid.coords["longitude"].values` has a size of 2161, and the increment is (180-(-180))/(2161-1) = 0.1666666...

Reply:
> Or we can calculate the increment from the minimum, maximum and length of the coordinates? Take the coordinates in [#1530 (comment)](https://github.com/GenericMappingTools/pygmt/pull/1530#discussion_r713752741) as an example, the array `grid.coords["longitude"].values` has a size of 2161, and the increment is (180-(-180))/(2161-1) = 0.1666666...

Implemented in d5d98c8c39f225c361a395a4d1ba275724dcdf7c.